SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialLatencyTracer.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "SpatialConstants.h"
+#include "Map.h"
+
+#if TRACE_LIB_ACTIVE
+#include "WorkerSDK/improbable/trace.h"
+#endif
+
+#include "SpatialLatencyTracer.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialLatencyTracing, Log, All);
+
+class AActor;
+class UFunction;
+class USpatialGameInstance;
+
+using TraceKey = int32;
+
+UCLASS()
+class SPATIALGDK_API USpatialLatencyTracer : public UObject
+{
+	GENERATED_BODY()
+
+public:
+
+	// Front-end exposed, allows users to register, start, continue, and end traces
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS", meta = (WorldContext = "WorldContextObject"))
+	static void RegisterProject(UObject* WorldContextObject, const FString& ProjectId);
+
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS", meta = (WorldContext = "WorldContextObject"))
+	static bool BeginLatencyTrace(UObject* WorldContextObject, const AActor* Actor, const FString& FunctionName, const FString& TraceDesc);
+
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS", meta = (WorldContext = "WorldContextObject"))
+	static bool ContinueLatencyTrace(UObject* WorldContextObject, const AActor* Actor, const FString& FunctionName, const FString& TraceDesc);
+
+	UFUNCTION(BlueprintCallable, Category = "SpatialOS", meta = (WorldContext = "WorldContextObject"))
+	static bool EndLatencyTrace(UObject* WorldContextObject);
+
+	static const TraceKey ActiveTraceKey = 0;
+	static const TraceKey InvalidTraceKey = -1;
+
+#if TRACE_LIB_ACTIVE
+
+	// Internal GDK usage, shouldn't be used by game code
+	static USpatialLatencyTracer* GetTracer(UObject* WorldContextObject);
+
+	bool IsValidKey(const TraceKey& Key);
```

Comment:
I believe this function is not thread-safe. It also doesn't seem to be useful enough to bother making it thread-safe (since there's code that checks key validity inside other functions as well). Should the function be removed?